packages/datetime/src/dateInput.tsx

```diff
@@ -387,28 +369,28 @@ export class DateInput extends AbstractPureComponent<IDateInputProps, IDateInput
 
     private handleInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
         const { valueString } = this.state;
-        const value = this.createMoment(valueString);
+        const value = stringToDate(valueString);
         if (
             valueString.length > 0 &&
             valueString !== this.getDateString(this.state.value) &&
-            (!value.isValid() || !this.isMomentInRange(value))
+            (!isValid(value) || !this.isDateInRange(value))
```

Comment:
Can we use `isDateValidAndInRange` here?

Reply:
Done